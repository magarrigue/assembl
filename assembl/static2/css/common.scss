html {
  font-size: 12px;  // the base for the rem unit (used in react-datepicker)
}

body {
  background-color: $page-bkg-grey;
  font-family: $font-family;
  font-size: $font-size-s;
  letter-spacing: $text-letter-spacing;
  overflow-y: scroll;
  overflow-x: hidden;
  -ms-overflow-style: scrollbar; // prevent transparent scrollbar in IE
}

a {
  color: $link-color;
  text-decoration: none;

  &:hover {
    color: $link-hover-color;
    text-decoration: none;
  }
}

input {
  &[type='text'] {
    box-shadow: 0 0 0 50px $input-txt-bkg-color inset;

    &:focus {
      box-shadow: 0 0 0 50px $input-txt-bkg-color inset;
    }
  }

  &[type='password'] {
    box-shadow: 0 0 0 50px $input-txt-bkg-color inset;

    &:focus {
      box-shadow: 0 0 0 50px $input-txt-bkg-color inset;
    }
  }
}

input:-moz-placeholder, textarea:-moz-placeholder {
  font-family: $font-family6;
}

input::-webkit-input-placeholder, textarea::-webkit-input-placeholder {
  font-family: $font-family6;
}

input.placeholder, textarea.placeholder {
  font-family: $font-family6;
}

.overflow-x {
  overflow-x: scroll;
}

.pre-wrap {
  white-space: pre-wrap;
}

.white {
  color: $white;
}

.grey {
  color: $grey2;
}

.black {
  color: $black;
}

.color {
  color: $first-color;
}

.color2 {
  color: $second-color;
}

.app-content {
  margin: $navbar-height 0 0;
}

.max-container {
  margin: 0 auto;
  max-width: $app-max-width;
}

.background-light {
  background-color: $page-bkg-light;
}

.background-grey {
  background-color: $page-bkg-grey;
}

.background-dark-grey {
  background-color: $page-bkg-dark-grey;
}

.background-dark {
  background-color: $page-bkg-dark;
}

.background-color {
  background-color: $first-color-light;
}

.background-img {
  background-position: top left;
  background-repeat: no-repeat;
  background-size: cover;
}

.content-section {
  padding: $margin-section 0;
}

.mobile-content-section {
  padding-top: 40px !important;
}

.content-margin {
  margin: -30px 0 0;
}

.inline {
  display: inline-block;
}

.left {
  float: left;
}

.right {
  float: right;
}

.center {
  text-align: center;
  margin-left: auto;
  margin-right: auto;
}

.justify {
  text-align: justify;
}

.border-top {
  border-top: $navbar-border;
}

.border-right {
  border-right: $navbar-border;
}

.border-left {
  border-left: $navbar-border;
}

.border-bottom {
  border-bottom: $navbar-border;
}

.border {
  border: $navbar-border;
}

.no-padding {
  padding: 0;
}

.no-margin {
  margin: 0;
}

.padding {
  padding: 15px;
}

.margin-xs {
  margin-top: 5px;
}

.margin-s {
  margin-top: 10px;
}

.margin-m {
  margin-top: 20px;
}

.margin-l {
  margin-top: 40px;
}

.margin-xl {
  margin-top: 60px;
}

.margin-xxl {
  margin-top: 80px;
}

.margin-xxxl {
  margin-top: 120px;
}

.margin-left-xxl {
  margin-left: 75px;
}

.margin-left-2 {
  margin-left: 2px;
}

.full-size {
  width: 100%;
}

.underline {
  border-bottom: $box-border;
  margin: 0 0 20px;
}

.uppercase {
  text-transform: uppercase;
}

.hidden {
  display: none;
}

.shown {
  display: block;
}

.col-centered {
  float: none;
  margin: 0 auto;
}

.clear {
  clear: both;
}

.relative {
  position: relative;
}

.pointer {
  cursor: pointer;
}

.no-pointer-events {
  pointer-events: none;
}

.form-label {
  color: $grey3;
  font-family: $font-family;
  font-size: $font-size-xs;
  padding: 0 0 0 5px;
}

.form-label-body {
  position: absolute;
  top: 131px;
}

.form-tree > .form-container > .form-tree-item {
  &::after,
  &::before {
    display: none;
  }
}

.form-tree {
  .form-container:first-child > .form-tree-item::before {
    display: none;
  }

  .form-container:first-child > .form-tree-item::after {
    border-radius: 5px 0 0 0;
  }

  .plus.form-tree-item:last-child::after,
  .form-container:last-child > .form-tree-item::after {
    display: none;
  }

  .plus.form-tree-item:last-child::before,
  .form-container:last-child > .form-tree-item::before {
    border-radius: 0 0 0 5px;
  }

  .form-container {
    .form-group {
      margin-bottom: 10px !important;
    }

    .form-tree-item {
      position: relative;

      &.plus {
        margin: initial;
        height: 15px;
        width: 15px;
        font-size: 12px;
      }
    }

    .form-tree-item::after {
      position: absolute;
      content: "";
      top: 18px;
      left: -15px;
      border-left: 1px solid #ccc;
      border-top: 1px solid #ccc;
      border-radius: 0 0 0 0;
      width: 15px;
      height: 100%;
    }

    .form-tree-item.plus::after {
      top: 8px;
    }

    .form-tree-item::before {
      content: "";
      position: absolute;
      top: -26px;
      left: -15px;
      border-left: 1px solid #ccc;
      border-bottom: 1px solid #ccc;
      border-radius: 0 0 0 0;
      width: 15px;
      height: 45px;
    }

    .form-tree-item.plus::before {
      height: 19px;
      top: -10px;
    }
  }

  .form-branch {
    margin-left: 25px;
    padding-right: 25px;
  }
}

.capitalized {
  text-transform: capitalize;
}

@media screen and (min-width: 992px) {
  .col-bottom {
    display: table-cell;
    float: none;
    vertical-align: bottom;
  }
}

@media screen and (max-width: 420px) {
  .content-section {
    padding: $margin-section-xs 0;
  }
}

strong {
  font-family: $font-family3;
}

#pDebug {
  display: none;
}

.cursor-wait {
  cursor: wait !important;
}

@media screen and (min-width: 1024px) {
  .page-body {
    margin: 0 90px 80px 90px;
  }
}

.small-hyphen-padding {
  padding: 0 5px;
}

.jumbotron {
  margin-top: 50px;
}

.flex {
  display: flex;
}

.center-flex {
  display: flex;
  justify-content: center;
}
