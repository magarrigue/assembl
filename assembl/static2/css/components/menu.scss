
.menu {
  padding: 0;
  list-style-type: none;

  li.menu-item {
    padding: 0 0 15px;

    a {
      color: $grey2;

      &:hover {
        color: $first-color;
      }
    }

    .arrow {
      color: $grey2;
      font-size: 6px;
      margin-right: 10px;
      margin-left: -19px;
      cursor: pointer;
    }

    .menu-item-title {
      display: flex;
      align-items: center;
    }

    .sub-menu {
      padding: 10px 0 0 10px;
      list-style-type: none;

      > li.menu-item:last-child {
        padding: 0 !important;
      }
    }

    &.active {
      > .menu-item-title {
        > .arrow, > a {
          color: $first-color;
        }
      }
    }
  }
}
