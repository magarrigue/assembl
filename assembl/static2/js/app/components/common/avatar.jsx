import React from 'react';
import { connect } from 'react-redux';
import { compose, graphql } from 'react-apollo';
import { Translate } from 'react-redux-i18n';
import { Link } from 'react-router';
import { NavDropdown, MenuItem } from 'react-bootstrap';

import { getContextual, get } from '../../utils/routeMap';
import UserQuery from '../../graphql/userQuery.graphql';
import withoutLoadingIndicator from './withoutLoadingIndicator';
import { browserHistory } from '../../router';

class ProfileIcon extends React.Component {
  constructor(props) {
    super(props);
    this.setCurrentView = this.setCurrentView.bind(this);
  }

  componentWillMount() {
    this.setCurrentView();
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.location) {
      this.setState({ next: nextProps.location });
    }
  }

  setCurrentView() {
    const { location } = this.props;
    this.setState({
      next: location
    });
  }

  render() {
    const { slug, connectedUserId } = this.props;
    const dropdownUser = (
      <div className="inline">
        <span className="assembl-icon-profil grey" />
        <span className="user-account"><Translate value="profile.panelTitle" /></span>
      </div>
    );
    return (
      <div className="right avatar">
        {!connectedUserId && (
          <Link to={`${getContextual('login', { slug: slug })}?next=${this.state.next}`}>
            <div className="connection">
              <Translate value="navbar.connection" />
            </div>
          </Link>
        )}
        {connectedUserId && (
          <ul className="dropdown-xs">
            <NavDropdown pullRight title={dropdownUser} id="user-dropdown">
              <MenuItem
                onClick={() => {
                  browserHistory.push(get('profile', { slug: slug, userId: connectedUserId }));
                }}
              >
                <Translate value="profile.panelTitle" />
              </MenuItem>
              <MenuItem href={`${getContextual('oldLogout', { slug: slug })}?next=${get('home', { slug: slug })}`}>
                <Translate value="navbar.logout" />
              </MenuItem>
            </NavDropdown>
          </ul>
        )}
      </div>
    );
  }
}

const mapStateToProps = ({ context, debate }) => ({
  slug: debate.debateData.slug,
  connectedUserId: context.connectedUserId,
  id: context.connectedUserIdBase64
});

export default compose(
  connect(mapStateToProps),
  graphql(UserQuery, {
    props: ({ data }) => {
      if (data.loading) {
        return { loading: true };
      }
      if (data.error) {
        return { error: data.error };
      }
      return {
        displayName: data.user.displayName
      };
    }
  }),
  withoutLoadingIndicator()
)(ProfileIcon);