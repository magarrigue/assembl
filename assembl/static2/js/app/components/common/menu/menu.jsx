// @flow
import * as React from 'react';
import classNames from 'classnames';

type Props = {
  className: string,
  parents: Array<number>,
  openedPath: Array<number>,
  children: React.Node,
  isSubMenu: boolean
};

type State = {
  openedItem: ?number
};

class Menu extends React.Component<Props, State> {
  static defaultProps = {
    isSubMenu: false,
    parents: [],
    openedPath: [],
    className: ''
  };

  static getDerivedStateFromProps(props: Props) {
    const { openedPath } = props;
    return {
      openedItem: openedPath ? openedPath[0] : null
    };
  }

  state = {
    openedItem: null
  };

  toggleMenuItem = (id: number) => {
    this.setState(prevState => ({ openedItem: prevState.openedItem !== id ? id : null }));
  };

  render() {
    const { className, isSubMenu, parents, openedPath, children } = this.props;
    const { openedItem } = this.state;
    const childrenElm = React.Children.map(children, (child, index) => {
      const spreadPath = openedItem !== null && openedPath[0] === openedItem;
      return React.cloneElement(child, {
        openedPath: spreadPath && openedPath.slice(1),
        parents: parents,
        id: index,
        openedItem: openedItem,
        toggle: this.toggleMenuItem
      });
    });
    return <ul className={classNames(className, { menu: !isSubMenu, 'sub-menu': isSubMenu })}>{childrenElm}</ul>;
  }
}

export default Menu;