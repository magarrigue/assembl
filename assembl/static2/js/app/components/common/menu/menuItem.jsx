// @flow
import * as React from 'react';
import classNames from 'classnames';

type Props = {
  id?: number,
  title: React.Node,
  className: string,
  parents: Array<number>,
  openedPath: Array<number>,
  children: React.Node,
  openedItem: ?number,
  toggle: ?(id: number | void) => void
};

const MenuItem = ({ id, parents, title, toggle, className, openedItem, openedPath, children }: Props) => {
  const active = openedItem === id;
  const childrenElm = React.Children.map(children, (child) => {
    const indexes = [...parents, id];
    return React.cloneElement(child, {
      parents: indexes,
      openedPath: openedPath,
      isSubMenu: true
    });
  });
  return (
    <li className={classNames(className, 'menu-item', { active: active })}>
      <div className="menu-item-title">
        {childrenElm ? (
          <span
            className={classNames('arrow', {
              'assembl-icon-up-open': active,
              'assembl-icon-down-open': !active
            })}
            onClick={toggle ? () => toggle(id) : null}
          />
        ) : null}
        {title}
      </div>
      {active && childrenElm ? childrenElm : null}
    </li>
  );
};

MenuItem.defaultProps = {
  id: 0,
  parents: [],
  openedPath: [],
  openedItem: null,
  className: '',
  toggle: null
};

export default MenuItem;