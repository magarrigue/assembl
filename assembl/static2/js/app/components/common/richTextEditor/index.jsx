// @flow
import RichTextEditor from './new';
// import RichTextEditor from './old';

export default RichTextEditor;