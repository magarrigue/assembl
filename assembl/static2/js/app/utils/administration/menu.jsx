// @flow
import * as React from 'react';

export type MenuItem = {
  title?: string,
  sectionId: string,
  component?: React.ComponentType<any>,
  subMenu?: {
    [key: string]: MenuItem
  }
};

export type Menu = {
  [key: string]: MenuItem
};

export function getAdminMenuSection(section: string, menu: Menu | null) {
  if (!menu) return null;
  const [index, ...subIndexes] = section.split('.');
  const item = menu[Object.keys(menu).filter(key => menu && menu[key].sectionId === index)[0]];
  const subMenu = item.subMenu;
  return subIndexes.length === 0 || !item ? item : getAdminMenuSection(subIndexes.join('.'), subMenu || null);
}