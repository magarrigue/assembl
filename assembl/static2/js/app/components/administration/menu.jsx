// @flow
import React from 'react';
import { Link } from 'react-router';
import { Translate } from 'react-redux-i18n';
import { get } from '../../utils/routeMap';
import { getDiscussionSlug } from '../../utils/globalFunctions';
import { type MenuItem } from '../../utils/administration/menu';
import PhaseMenu from './phaseMenu';
import { ADMIN_MENU } from '../../constants';

type Props = {
  requestedPhase: string,
  i18n: { locale: string },
  timeline: Timeline
};

class Menu extends React.PureComponent<Props> {
  renderMenuItem = (
    id: string,
    menuItem: MenuItem,
    slug: { slug: string | null },
    rootSection: string = '',
    isRoot: boolean = true
  ) => {
    const { requestedPhase } = this.props;
    const { title, sectionId, subMenu } = menuItem;
    const sectionIndex = rootSection ? `${rootSection}.${sectionId}` : sectionId;
    const sectionQuery = sectionId ? `?section=${sectionIndex}` : '';
    const subMenuIds = subMenu ? Object.keys(subMenu) : [];
    const newRootSection = !isRoot ? sectionIndex : '';
    return (
      <li key={sectionIndex} className={isRoot ? 'menu-item' : ''}>
        <Link to={`${get('administration', slug)}/${id}${sectionQuery}`} activeClassName="active">
          <Translate value={title} />
        </Link>
        {subMenu && subMenuIds.length > 0 ? (
          <ul className={requestedPhase === id ? 'shown admin-menu2' : 'hidden admin-menu2'}>
            {subMenuIds.map((subKey) => {
              const subMenuItem = subMenu[subKey];
              return this.renderMenuItem(id, subMenuItem, slug, newRootSection, false);
            })}
          </ul>
        ) : null}
      </li>
    );
  };

  render() {
    const slug = { slug: getDiscussionSlug() };
    const { timeline } = this.props;
    const { requestedPhase, i18n: { locale } } = this.props;
    return (
      <ul className="admin-menu">
        {Object.keys(ADMIN_MENU).map(key => this.renderMenuItem(key, ADMIN_MENU[key], slug))}
        {timeline
          ? timeline.map((phase, index) => (
            <PhaseMenu
              key={phase.id}
              slug={slug}
              index={index}
              phase={phase}
              isActive={phase.identifier === requestedPhase}
              locale={locale}
            />
          ))
          : null}
      </ul>
    );
  }
}

export default Menu;