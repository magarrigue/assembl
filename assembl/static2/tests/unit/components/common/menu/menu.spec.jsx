import React from 'react';
import renderer from 'react-test-renderer';

import { Menu, MenuItem } from '../../../../../js/app/components/common/menu';

describe('Menu component', () => {
  it('should match Menu snapshot', () => {
    const rendered = renderer
      .create(
        <Menu>
          <MenuItem title={<span>Foo</span>}>
            <Menu>
              <MenuItem title={<span>Sub Foo</span>} />
            </Menu>
          </MenuItem>
          <MenuItem title={<span>Bar</span>} />
        </Menu>
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
  it('should match opened Menu snapshot', () => {
    const rendered = renderer
      .create(
        <Menu openedPath={[0, 0]}>
          <MenuItem title={<span>Foo</span>}>
            <Menu>
              <MenuItem title={<span>Sub Foo</span>} />
            </Menu>
          </MenuItem>
          <MenuItem title={<span>Bar</span>} />
        </Menu>
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
});