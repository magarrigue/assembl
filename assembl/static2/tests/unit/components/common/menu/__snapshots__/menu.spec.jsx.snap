// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component should match Menu snapshot 1`] = `
<ul
  className="menu"
>
  <li
    className="menu-item"
  >
    <div
      className="menu-item-title"
    >
      <span
        className="arrow assembl-icon-down-open"
        onClick={[Function]}
      />
      <span>
        Foo
      </span>
    </div>
  </li>
  <li
    className="menu-item"
  >
    <div
      className="menu-item-title"
    >
      <span>
        Bar
      </span>
    </div>
  </li>
</ul>
`;

exports[`Menu component should match opened Menu snapshot 1`] = `
<ul
  className="menu"
>
  <li
    className="menu-item active"
  >
    <div
      className="menu-item-title"
    >
      <span
        className="arrow assembl-icon-up-open"
        onClick={[Function]}
      />
      <span>
        Foo
      </span>
    </div>
    <ul
      className="sub-menu"
    >
      <li
        className="menu-item active"
      >
        <div
          className="menu-item-title"
        >
          <span>
            Sub Foo
          </span>
        </div>
      </li>
    </ul>
  </li>
  <li
    className="menu-item"
  >
    <div
      className="menu-item-title"
    >
      <span>
        Bar
      </span>
    </div>
  </li>
</ul>
`;
